examples/vision/siamesenetwork.py

```diff
@@ -0,0 +1,417 @@
+"""
+Title: Siamese Network with Triplet Loss
+Author: [Hazem Essam](https://twitter.com/hazemessamm) and [Santiago L. Valdarrama](https://twitter.com/svpino)
+Date created: 2021/03/13
+Last modified: 2021/03/22
+Description: Siamese network with tf.data and custom training loop.
+"""
+
+
+"""
+TODO: Still need to work on this.
+
+# Introduction
+
+[Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network) is used to solve
+many problems like detecting question duplicates, face recognition by comparing the
+similarity of the inputs by comparing their feature vectors.
+First we need to have a dataset that contains 3 Images, 2 are similar and 1 is different,
+they are called Anchor image, Positive Image and Negative image respectively, we need to
+tell the network that the anchor image and the positive image are similar, we also need
+to tell it that the anchor image and the negative image are NOT similar, we can do that
+by the Triplet Loss Function.
+Triplet Loss function:
+L(Anchor, Positive, Negative) = max((distance(f(Anchor), f(Positive)) -
+distance(f(Anchor), f(Negative)))**2, 0.0)
```

Comment:
Suggestion:

```suggestion
The triplet loss function is measured as follows:

L(Anchor, Positive, Negative) = max((distance(f(Anchor), f(Positive)) -
distance(f(Anchor), f(Negative)))**2, 0.0)

```

(maybe it'll look better in Latex?)